src/views/design/ComponentDetail/index.tsx

```diff
@@ -598,14 +597,12 @@ const ComponentDetail = withStyles(styles)(
                     mandatory={mandatory}
                     isCreateParameter={isAddingParameter}
                     onEdit={(newParameter) => {
-                        const newParameters = [...newComponentDetail.parameters];
-                        if (!isAddingParameter) {
-                            newParameters[editParameterIndex] = newParameter;
-                        }
+                        const newParameters = isAddingParameter
+                            ? [...newComponentDetail.parameters, newParameter] : [...newComponentDetail.parameters];
+                        newParameters[editParameterIndex] = newParameter;
+                        const orderedParameters = orderParameter(newParameters, matchingComponentType);
```

Comment:
this doesn't seem to be correct.
Either you're adding a parameter and then you add the new parameter to the existing ones OR you're editing an existing  parameter and then you set the parameter at editParameterIndex to the new parameter. 
Not a combination of both